File: day2/imo/submissions/partially_accepted/template.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int N, M, K;
    cin >> N >> M >> K;

    vector<vector<int>> a(N, vector<int>(M));
    for (int i = 0; i < N; ++i) {
        for (int j = 0; j < M; ++j) {
            cin >> a[i][j];
        }
    }

    vector<vector<int>> scores(0);
    for (int i = 0; i < N; i++) {
	int ones = 0;
        for (int j = 0; j < M; j++) {
	    if (a[i][j] == 1) ones++;
	}
	scores.push_back({ones, -i});
    }
    sort(scores.begin(), scores.end());
    int gaps = 0;
    for (int i = 0; i < N-1; i++) {
        if (scores[i+1][1] < scores[i][1]) gaps++;
    }

    int ans = N*M-M+gaps;

    cout << ans << endl;

    return 0;
}

File: day2/imo/submissions/partially_accepted/template.py
#!/usr/bin/env python3
N, M, K = map(int, input().split())

a = [
    list(map(int, input().split()))
    for i in range(N)
]

scores = []
for i in range(N):
    ones = 0
    for j in range(M):
        if a[i][j] == 1:
            ones += 1
    scores.append((ones, -i))

scores.sort()
gaps = 0
for i in range(N-1):
    if scores[i+1][1] < scores[i][1]:
        gaps += 1

ans = N*M-M+gaps

print(ans)
